packages/rust/armonik/src/client/agent.rs

```diff
@@ -0,0 +1,402 @@
+}
+
+impl<T> AgentClient<T>
+where
+    T: tonic::client::GrpcService<tonic::body::BoxBody>,
+    T::Error: Into<tonic::codegen::StdError>,
+    T::ResponseBody: tonic::codegen::Body<Data = tonic::codegen::Bytes> + Send + 'static,
+    <T::ResponseBody as tonic::codegen::Body>::Error: Into<tonic::codegen::StdError> + Send,
+{
+    pub fn with_channel(channel: T) -> Self {
+        Self {
+            inner: v3::agent::agent_client::AgentClient::new(channel),
+        }
+    }
+
+    /// Create the metadata of multiple results at once.
+    /// Data have to be uploaded separately.
+    pub async fn create_results_metadata(
+        &mut self,
+        token: impl Into<String>,
+        session_id: impl Into<String>,
+        names: impl std::iter::IntoIterator<Item = impl Into<String>>,
+    ) -> Result<HashMap<String, ResultMetaData>, super::RequestError> {
+        Ok(self
+            .call(create_results_metadata::Request {
+                communication_token: token.into(),
+                results: names.into_collect(),
+                session_id: session_id.into(),
+            })
+            .await?
+            .results)
+    }
+
+    /// Create multiple results with data included in the request.
+    pub async fn create_results(
+        &mut self,
+        token: impl Into<String>,
+        session_id: impl Into<String>,
+        results: impl std::iter::IntoIterator<Item = (impl Into<String>, impl Into<Vec<u8>>)>,
+    ) -> Result<HashMap<String, ResultMetaData>, super::RequestError> {
+        Ok(self
+            .call(create_results::Request {
+                communication_token: token.into(),
+                results: results
+                    .into_iter()
+                    .map(|(name, data)| (name.into(), data.into()))
+                    .collect(),
+                session_id: session_id.into(),
+            })
+            .await?
+            .results)
+    }
+
+    /// Create tasks metadata and submit task for processing.
+    pub async fn submit(
+        &mut self,
+        token: impl Into<String>,
+        session_id: impl Into<String>,
+        task_options: Option<TaskOptions>,
+        items: impl IntoIterator<Item = submit_tasks::RequestItem>,
+    ) -> Result<Vec<submit_tasks::ResponseItem>, super::RequestError> {
+        Ok(self
+            .call(submit_tasks::Request {
+                communication_token: token.into(),
+                session_id: session_id.into(),
+                task_options,
+                items: items.into_collect(),
+            })
+            .await?
+            .items)
+    }
+
+    pub async fn create_tasks(
+        &mut self,
+        request: impl Stream<Item = create_tasks::Request> + Send + 'static,
+    ) -> Result<Vec<create_tasks::Status>, super::RequestError> {
+        let response = self.call(request).await?;
+
+        match response {
+            create_tasks::Response::Status {
+                communication_token: _,
```

Comment:
I think we need a token here

Reply:
No, I don't return the token. Here it is the response.